sdk/tables/Azure.Data.Tables/src/TableEntity.cs

```diff
@@ -235,5 +236,56 @@ private static void EnforceType(Type requestedType, Type givenType)
                     $"Cannot return {requestedType} type for a {givenType} typed property."));
             }
         }
+
+        /// <summary>
+        /// Performs type coercion for numeric types.
+        /// Integers will be coerced to long or double.
+        /// Double and long values will accept integers values but remain double or long.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">The type coercion is not allowed.</exception>
+        private static object CoerceType(object existingValue, object newValue)
+        {
+            if (!existingValue.GetType().IsAssignableFrom(newValue.GetType()))
+            {
+                object convertedValue = existingValue switch
+                {
+                    int _ => newValue switch
+                    {
+                        // All int values can be doubles
+                        double newDoubleValue =>  newDoubleValue,
```

Comment:
do we support deciman?

Reply:
You are unnecessarily boxing the newDoubleValue here. Returning newValue would avoid an extra allocation.

Reply:
We don't explicitly support decimal. For bytes, we're not allowing implicit conversion.